src/Components/Web.JS/src/Platform/WebAssemblyProgressReporter.ts

```diff
@@ -0,0 +1,21 @@
+import { WebAssemblyProgressService } from "./WebAssemblyProgressService";
+
+export class WebAssemblyProgressReporter {
+    static setProgress(resourcesTotal: number, resourcesLoaded: number): void {
+        const circle = document.querySelector('.progress') as SVGCircleElement;
+        const circumference = 2 * Math.PI * circle.r.baseVal.value;
+        const ring = circumference - resourcesLoaded / resourcesTotal * circumference;
+        circle.style.strokeDasharray = circumference.toString() + " " + circumference.toString();
```

Comment:
I'm not too familiar with this, so I may not be on the right track here; but if these two values are the same, do the values matter at all? [Looks like](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-dasharray) if the values are the same it is equivalent to a "dashed line" with equal spacing? Again, not really familiar with this so I could be totally off.

Reply:
The first value of `strokeDashArray` creates a dash that is the same length as the circumference of the ring. The second value creates the spacing so that only one dash (representing the progress) is visible and encompasses the entire circumference.